src/components/Exchange/SwapBox.js

```diff
@@ -1786,6 +1750,39 @@ export default function SwapBox(props) {
     }
   }
 
+  const fromUsdMinAfterFees = fromUsdMin?.sub(swapFees ?? 0).sub(positionFee ?? 0) || bigNumberify(0);
+  const liquidationPrice = getLiquidation({
+    isLong,
+    size: hasExistingPosition ? existingPosition.size.add(toUsdMax || 0) : toUsdMax ?? bigNumberify(0),
+    collateral: hasExistingPosition
+      ? existingPosition.collateralAfterFee.add(fromUsdMinAfterFees)
+      : fromUsdMinAfterFees ?? bigNumberify(0),
+    averagePrice: nextAveragePrice ?? bigNumberify(0),
+  });
+
+  const existingLiquidationPrice = existingPosition
+    ? getLiquidation({
+        isLong: existingPosition.isLong,
+        size: existingPosition.size,
+        collateral: existingPosition.collateral,
+        averagePrice: existingPosition.averagePrice,
+        fundingFee: existingPosition.fundingFee,
+      })
+    : undefined;
+  let displayLiquidationPrice = liquidationPrice ? liquidationPrice : existingLiquidationPrice;
+
+  if (hasExistingPosition) {
+    leverage = getLeverageNew({
+      size: existingPosition.size.add(toUsdMax || 0),
+      collateral: existingPosition.collateralAfterFee.add(fromUsdMinAfterFees),
+      delta: nextDelta,
+      hasProfit: nextHasProfit,
+      includeDelta: savedIsPnlInLeverage,
+    });
+  } else if (hasLeverageOption) {
+    leverage = bigNumberify(parseInt(leverageOption * BASIS_POINTS_DIVISOR));
```

Comment:
just to confirm, why not calculate `leverage` based on `toUsdMax` and `fromUsdMinAfterFees` if hasLeverageOption==false?